### ISSUE TITLE
Default for undefined booleans without default (azurerm_storage_account/enable_blob_encryption)

### ISSUE THREAD

**Issue by anoff** [terraform-providers/terraform-provider-azurerm#1028]
### Terraform Version
Terraform v0.11.5
+ provider.azurerm v1.3.0

### Affected Resource(s)
- azurerm_storage_account

### Terraform Configuration Files
```hcl
resource "azurerm_storage_account" "backend" {
  name                     = "mybackend"
  resource_group_name      = "${azurerm_resource_group.backend.name}"
  location                 = "${azurerm_resource_group.backend.name.location}"
  account_tier             = "Standard"
  account_kind             = "StorageV2"
  account_replication_type = "GRS"
}
```

I was wondering how TF handles boolean values that are set to `computed` if there is no `default` present.
The one I am especially interested in is `enable_blob_encryption` [option](https://www.terraform.io/docs/providers/azurerm/r/storage_account.html#enable_blob_encryption). I assume since the [code](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/data_source_storage_account.go#L64) does not specify a `default: true` it will go to [0 -> false](https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_data.go#L518).

However since Azure enforces encryption on all storage accounts per default I wonder how Terraform behaves in this scenario.

**Comment by MarcoMol** [comment]
+1
I am noticing the behaviour where terraform defaults to False - while the azure platform defaults to true.

**Comment by katbyte** [comment]
Hello @anoff,

While the code does not specify a default of `true`, it only passing a value of `true` or `false` to the API if the property is set that then defaults to true. However i understand the confusion so have opened #1380 to explicitly set the default to true.

@MarcoMol, could you provide some example terraform where the provider ends up defaulting to `false`? thank you

**Comment by MarcoMol** [comment]
Hi @katbyte,

of course, here goes:
```
resource "azurerm_storage_account" "id" {
  account_tier             = "Standard"
  resource_group_name      = "${azurerm_resource_group.name.name}"
  name                     = "name"
  account_replication_type = "GRS"
  location                 = "westeurope"
  account_kind             = "StorageV2"
}
```

the other storage account is the same, but LRS.

**Comment by katbyte** [comment]
Hi @MarcoMol,

Thank you for providing the config, updated #1380 as a result and this should be fixed in `1.7.0`.

**Comment by katbyte** [comment]
Hey @anoff @MarcoMol,

We just released 1.7 today that included the fix for this issue, hope it solves it for you two!

**Comment by anoff** [comment]
@katbyte thanks a lot for keeping us updated
